wp-lazy-loading.php

```diff
@@ -141,14 +139,36 @@ function wp_add_lazy_load_attributes( $content, $context = null ) {
 
 	return preg_replace_callback(
 		'/<img\s[^>]+>/',
-		function( array $matches ) {
+		function( array $matches ) use( $content, $context ) {
 			if ( ! preg_match( '/\sloading\s*=/', $matches[0] ) ) {
-				return str_replace( '<img', '<img loading="lazy"', $matches[0] );
+				$tag_html = $matches[0];
+
+				/**
+				 * Filters the `loading` attribute value. Default `lazy`.
+				 *
+				 * Returning `false` or an empty string will not add the attribute.
+				 * Returning `true` will add the default value.
+				 *
+				 * @since (TBD)
+				 *
+				 * @param string $default The filtered value, defaults to `lazy`.
+				 * @param string $tag_html The tag's HTML.
+				 * @param string $content The HTML containing the image tag.
+				 * @param string $context Optional. Additional context. Defaults to `current_filter()`.
+				 */
+				$value = apply_filters( 'wp_set_image_loading_attr', 'lazy', $tag_html, $content, $context );
+
+				if ( $value ) {
+					if ( ! in_array( $value, array( 'lazy', 'eager' ), true ) ) {
```

Comment:
```suggestion
					if ( ! is_string( $value ) ) {
```

Reply:
Don't think this is a good idea. If we wanted plugins to be able to add "unrestricted HTML" there, better to filter the whole `<img>` tag (as it was previously). If we change this to `! is_string( $value )` that string could be an arbitrary amount of HTML.

Reply:
We would need to use `esc_attr()` and it would be okay. Anyway, this is a minor concern here.

Reply:
Yeah, `esc_attr()` would work but adds some overhead. Ensuring the attribute value is "standard" seems to make most sense but agree this is a minor concern.